File: c-diamond/diamond.c
#include <stdio.h>
#include <stdlib.h>


void print_line(int space, int star){
    for (int i = 0; i < space; i++) {
        printf(" ");
    }
    for (int i = 0; i < star; i++){
        printf("*");
    }
    printf("\n");
}

void print_diamond(int n){
    if (n == 1){
        printf("*\n");
    }
    else {
        int spaces = n-1;
        int stars = 1;
        do {
            //calculate stars, start with 1, increment by 2.
            //calculate spaces, start with n-1, decrement by 1.
        
            print_line(spaces, stars);
            spaces --; 
            stars += 2;
        } while (stars < n+n);
        spaces = 0;
        stars = n+n-1;
        do {
            spaces ++;
            stars = stars - 2;
            print_line(spaces, stars);
        } while (spaces < n-1);
    }

}

File: diamond.py
#include <stdio.h>
#include <stdlib.h>

def print_diamond(n):
    if (n == 1):
        print("*\n")
    else:
        spaces = []
        for i in range(n):
            spaces.append(' ')
        spaces[n-1]='*'
        spaces.append('\0')         #spaces[n] = '\0'
        print("%s\n", spaces)
        while (spaces[0] != '*'):
            spaces.append('*', [n-1-i])  #spaces[n-1-i] = '*'
            spaces.append('*', [n+i+1]) #spaces[n+i+1] = '*'
            print("%s\n", spaces)
            i+=1
        while (spaces[n] != '\n'): 
            pass

print_diamond(3)